crates/ui/src/dock/tiles_panel.rs

```diff
@@ -0,0 +1,728 @@
+        cx: &mut ViewContext<Self>,
+    ) {
+        self.panels.push(TilesItem {
+            panel: panel.clone(),
+            bounds,
+            z_index,
+        });
+
+        cx.emit(PanelEvent::LayoutChanged);
+        cx.notify();
+    }
+
+    // Check if a bounds is occluded by higher z-index panels
+    fn is_bounds_occluded(&self, current_index: usize, bounds: &Bounds<Pixels>) -> bool {
+        let current_z_index = self.panels[current_index].z_index;
+        self.panels.iter().enumerate().any(|(index, item)| {
+            index != current_index
+                && item.z_index > current_z_index
+                && item.bounds.intersects(bounds)
+        })
+    }
+
+    // Find the panel at a given position, considering z-index
+    fn find_panel_at_position(&self, position: Point<Pixels>) -> Option<(usize, &TilesItem)> {
+        let adjusted_position = position - self.bounds.origin;
+        let mut panels_with_indices: Vec<(usize, &TilesItem)> =
+            self.panels.iter().enumerate().collect();
+        panels_with_indices.sort_by(|a, b| b.1.z_index.cmp(&a.1.z_index));
+
+        for (index, item) in panels_with_indices {
+            let extended_bounds = Bounds::new(
+                item.bounds.origin,
+                Size {
+                    width: item.bounds.size.width + px(HALF_HANDLE_SIZE),
+                    height: item.bounds.size.height + px(HALF_HANDLE_SIZE),
+                },
+            );
+
+            if extended_bounds.contains(&adjusted_position) {
+                return Some((index, item));
+            }
+        }
+
+        None
+    }
+
+    /// Bring the panel of target_index to front by updating its z_index
+    fn bring_panel_to_front(&mut self, target_index: Option<usize>) {
+        if let Some(index) = target_index {
+            let max_z_index = self
+                .panels
+                .iter()
+                .map(|item| item.z_index)
+                .max()
+                .unwrap_or(0);
+
+            if let Some(item) = self.panels.get_mut(index) {
+                if item.z_index == max_z_index {
+                    return;
+                }
+                item.z_index = (max_z_index + 1) % usize::MAX;
+            }
+        }
+    }
+}
+
+fn round_to_nearest_ten(value: Pixels) -> Pixels {
+    px((value.0 / 10.0).round() * 10.0)
+}
+
+fn round_point_to_nearest_ten(point: Point<Pixels>) -> Point<Pixels> {
+    Point::new(round_to_nearest_ten(point.x), round_to_nearest_ten(point.y))
+}
+
+impl FocusableView for TilesPanel {
+    fn focus_handle(&self, _cx: &AppContext) -> FocusHandle {
+        self.focus_handle.clone()
+    }
+}
+impl EventEmitter<PanelEvent> for TilesPanel {}
+impl EventEmitter<DismissEvent> for TilesPanel {}
+
+impl Render for TilesPanel {
+    fn render(&mut self, cx: &mut ViewContext<Self>) -> impl IntoElement {
+        let entity_id = cx.entity_id();
+        let view = cx.view().clone();
+        let border = cx.theme().border;
```

Comment:
Avoid define a color variable, just use `cx.theme().border` in place where you want.